diff --git a/player/lua/console.lua b/player/lua/console.lua
index 170f41b..1c984ea 100644
--- a/player/lua/console.lua
+++ b/player/lua/console.lua
@@ -121,13 +121,13 @@ local function get_font()
         return
     end
 
-    -- Pick a better default font for Windows and macOS
+    -- Pick a better font for Windows and macOS
     if platform == 'windows' then
         return 'Consolas'
     end
 
     if platform == 'darwin' then
-        return 'Menlo'
+        return 'SF Mono'
     end
 
     return 'monospace'
